src/Resolvers/Dig.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace RemotelyLiving\PHPDNS\Resolvers;
+
+use RemotelyLiving\PHPDNS\Entities\DNSRecordCollection;
+use RemotelyLiving\PHPDNS\Entities\DNSRecordType;
+use RemotelyLiving\PHPDNS\Entities\Hostname;
+use RemotelyLiving\PHPDNS\Factories\SpatieDNS;
+use RemotelyLiving\PHPDNS\Mappers\Dig as DigMapper;
+use RemotelyLiving\PHPDNS\Resolvers\Exceptions\QueryFailure;
+
+class Dig extends ResolverAbstract
+{
+    private const SUPPORTED_QUERY_TYPES = [
+        DNSRecordType::TYPE_A,
+        DNSRecordType::TYPE_AAAA,
+        DNSRecordType::TYPE_CNAME,
+        DNSRecordType::TYPE_NS,
+        DNSRecordType::TYPE_SOA,
+        DNSRecordType::TYPE_MX,
+        DNSRecordType::TYPE_SRV,
+        DNSRecordType::TYPE_TXT,
+        DNSRecordType::TYPE_CAA,
+        DNSRecordType::TYPE_NAPTR,
+    ];
+
+    /**
+     * @var \RemotelyLiving\PHPDNS\Factories\SpatieDNS
+     */
+    private $spatieDNSFactory;
+
+    /**
+     * @var \RemotelyLiving\PHPDNS\Mappers\Dig
+     */
+    private $mapper;
+
+    /**
+     * @var \RemotelyLiving\PHPDNS\Entities\Hostname|null
+     */
+    private $nameserver;
+
+    public function __construct(
+        SpatieDNS $spatieDNSFactory = null,
+        DigMapper $mapper = null,
+        Hostname $nameserver = null
+    ) {
+        $this->spatieDNSFactory = $spatieDNSFactory ?? new SpatieDNS();
+        $this->mapper = $mapper ?? new DigMapper();
+        $this->nameserver = $nameserver;
+    }
+
+    protected function doQuery(Hostname $hostname, DNSRecordType $type): DNSRecordCollection
+    {
+        if (!self::isSupportedQueryType($type)) {
```

Comment:
fail quietly or maybe even throwing here? It gets caught in the parent and logged as a failure.